core/src/main/java/org/arquillian/cube/impl/client/container/ContainerConfigurationController.java

```diff
@@ -64,15 +64,16 @@ public void remapContainer(@Observes BeforeSetup event, CubeRegistry cubeRegistr
         Object newConfigurationInstance = configurationClass.newInstance();
 
         for (PropertyDescriptor configurationClassPortField : configurationClassPortFields) {
+            // Move it outside if condition below, if there is a group of containers.
+            int containerPort = getDefaultPortFromConfigurationInstance(newConfigurationInstance,
```

Comment:
I guess it will always enter in the first for loop.
I am not sure if we need to move it to the another for, because if there is no port to override also there is no address too.

Probably by adding this same if condition ( if (mappingForPort != null)  ) on the second for loop might be enough.

Does it make sense?